mie2c/e2c.py

```diff
@@ -356,23 +362,102 @@ def predict(self, X, U):
         x_next_dec = self.decoder.eval()(z_next_pred)
         return x_next_dec
 
+
 def compute_loss(x_dec, x_next_dec, x_next_pred_dec,
                  x, x_next,
-                 Qz, Qz_next, Qz_next_pred):
-
-    # # l2_norm
-    # x_reconst_loss = (x_dec - x).pow(2).sum(dim=[1,2,3])
-    # x_next_reconst_loss = (x_next_pred_dec - x_next).pow(2).sum(dim=[1,2,3])
-
-    # cross entropy loss
-    x_reconst_loss = F.binary_cross_entropy(nn.Sigmoid()(x_dec), nn.Sigmoid()(x), reduction='sum')
-    x_next_reconst_loss = F.binary_cross_entropy(nn.Sigmoid()(x_next_pred_dec), nn.Sigmoid()(x_next), reduction='sum')
+                 Qz, Qz_next, Qz_next_pred,
+                 use_l2=False):
+    if use_l2:
+        x_reconst_loss = (x_dec - x).pow(2).sum(dim=[1,2,3])
+        x_next_reconst_loss = (x_next_pred_dec - x_next).pow(2).sum(dim=[1,2,3])
+    else:
+        bce_loss = nn.BCELoss(reduction='mean')
+        x_reconst_loss = bce_loss(torch.clamp(x_dec, 0., 1.), x)
```

Comment:
Shouldn't _clamp_ be applied to x and x_next to ensure the image data is normalized?

Reply:
I think similar to last comment, I'm assuming that your dataset already comes in as 0-1 float32's. That's what it is for the Benchmark notebook. Is it different for the cartpole?

Reply:
Cartpole is currently not being preprocessed, but I can make the data processing consistent once this gets approved. However, why would _x_dec_ require a _clamp_ here? The output of the decoder is already normalized 0-1.

Reply:
you are correct (fixed)